controllers/managedocs_controller.go

```diff
@@ -1067,6 +1085,40 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) getPrometheusRemoteWriteSpec() (*promv1.RemoteWriteSpec, error) {
+	if err := r.get(r.rhobsSecret); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, fmt.Errorf("rhobs secret missing: %v", err)
+		} else {
+			return nil, fmt.Errorf("unable to fetch rhobs secret: %v", err)
+		}
+	}
+	rhobsSecretData := r.rhobsSecret.Data
+	if _, found := rhobsSecretData[rhobsConfigIdName]; !found {
+		return nil, fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsConfigIdName)
+	}
+
+	if _, found := rhobsSecretData[rhobsConfigSecretName]; !found {
+		return nil, fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsConfigSecretName)
+	}
+
+	rhobsAudience, found := rhobsSecretData["rhobs-audience"]
+	if !found {
+		return nil, fmt.Errorf("rhobs secret does not contain a value for key rhobs-audience")
+	}
+
+	remoteWriteSpec := templates.PrometheusRemoteWriteSpecTemplate.DeepCopy()
+	remoteWriteSpec.URL = r.RHOBSEndpoint
+	remoteWriteSpec.OAuth2.ClientID.Secret.LocalObjectReference.Name = r.RHOBSSecretName
+	remoteWriteSpec.OAuth2.ClientID.Secret.Key = rhobsConfigIdName
+	remoteWriteSpec.OAuth2.ClientSecret.LocalObjectReference.Name = r.RHOBSSecretName
+	remoteWriteSpec.OAuth2.ClientSecret.Key = rhobsConfigSecretName
+	remoteWriteSpec.OAuth2.TokenURL = r.RHSSOTokenEndpoint
+	remoteWriteSpec.OAuth2.EndpointParams["audience"] = string(rhobsAudience)
```

Comment:
Why are we returning a remote write spec instead of filling-in the one we get from the template?

Reply:
Done